code/modules/fabrication/_fabricator.dm

```diff
@@ -97,6 +99,10 @@
 				var/decl/material/reg = mat
 				stored_substances_to_names[mat] = lowertext(initial(reg.name))
 
+	if(prefilled)
```

Comment:
This would be better as an override of Initialize on a mapped subtype imo.

Reply:
To be clear, while maybe I have minor quibbles on implementation, I agree that this should be done using subtypes.